felix/iptables/table.go

```diff
@@ -314,7 +314,7 @@ func NewTable(
 	ipVersion uint8,
 	hashPrefix string,
 	iptablesWriteLock sync.Locker,
-	featuredetector *environment.FeatureDetector,
+	featuredetector *environment.FeatureDetectorImpl,
```

Comment:
I've flipped the naming so that the interface has the suffix. I think making concrete types private is a bit of an anti-pattern in Go; you're intended to define your interfaces locally for mocking but it got quite messy with the dozens of places I needed this interface.